pkg/apis/camel/v1/integration_types_support.go

```diff
@@ -255,6 +255,15 @@ func (in *SourceSpec) InferLanguage() Language {
 	return ""
 }
 
+// SetOperatorID sets the given operator id as an annotation
+func (in *Integration) SetOperatorID(operatorID string) {
+	if in.Annotations == nil {
+		in.Annotations = make(map[string]string)
+	}
+
+	in.Annotations[OperatorIDAnnotation] = operatorID
+}
+
```

Comment:
Duplicated code. We may have a single method with that logic, and call it from the various entities